docs/topics/request-response.rst

```diff
@@ -385,6 +385,49 @@ The meta key is used set retry times per request. When initialized, the
 :reqmeta:`max_retry_times` meta key takes higher precedence over the
 :setting:`RETRY_TIMES` setting.
 
+
+.. _topics-cancel-request:
+
+Cancelling a request
+====================
+
+Raising a :exc:`~scrapy.exceptions.CancelRequest` exception from a
+:class:`~scrapy.signals.bytes_received` signal handler will stop the
+download of a given request. See the following example::
+
+    import scrapy
+
+
+    class CancelSpider(scrapy.Spider):
+        name = "cancel"
+        start_urls = ["https://docs.scrapy.org/en/latest/"]
+
+        @classmethod
+        def from_crawler(cls, crawler):
+            spider = super().from_crawler(crawler)
+            crawler.signals.connect(spider.handler, signal=scrapy.signals.bytes_received)
+            return spider
+
+        def parse(self, response):
+            # 'last_chars' show that the full response was not downloaded
+            yield {"len": len(response.text), "last_chars": response.text[-40:]}
+
+        def handler(self, data, request, spider):
+            if b"some text that I need" not in data:  # data is a 'bytes' object
```

Comment:
This example is unfortunately not correct: "some text" + "that I need" can be in different chunks.

Reply:
haha fair enough. I initially had `if some_condition:` but I got concerned about someone reporting that `some_condition` raised a `NameError`.

Reply:
That's the great thing about DDD (docs driven development) - you try writing an usage example, it becomes complicated, and you then extract some of these complexity to helpers and add them to a library you're developing. So I think a real example could be more useful for users; they're 100% going to make the same mistake.